File: CPP/Program/Car.cpp
// Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once
// Memasukan library yang digunakan, termasuk kelas induk (kelas Vehicle) dari kelas Car
#include <iostream>
#include <string>
#include "Vehicle.cpp"

//deklarasi namespace
using namespace std;

//kelas car yang terdiri dari jumlah kursi mobil dan jumlah pintu mobil
//kelas car merupakn turunan dari kelas induk Vehicle
class Car : public Vehicle {
private:
    // Atribut yang digunakan pada kelas Car terdiri dari asal jumlah kursi dan jumlah pintu
    int jumlahKursi;
    int jumlahPintu;

public:
    //constructor kosong
    Car(){ 

    }

    //constructor yang diisi parameter
    Car(int jumlahKursi, int jumlahPintu, string platNomor, string merk, string tahunProduksi, string warna)
    : Vehicle(platNomor, merk, tahunProduksi, warna), jumlahKursi(jumlahKursi), jumlahPintu(jumlahPintu) {}

    //membuat getter setter
    void setJumlahKursi(int jumlahKursi) {
        this->jumlahKursi = jumlahKursi;
    }

    void setJumlahPintu(int jumlahPintu) {
        this->jumlahPintu = jumlahPintu;
    }

    int getJumlahKursi() {
        return jumlahKursi;
    }

    int getJumlahPintu() {
        return jumlahPintu;
    }

    //membuat fungsi untuk menampilkan jumlah kursi dan jumlah pintu
    void display() const override {
        Vehicle::display();
        cout << "Jumlah Kursi: " << jumlahKursi << endl;
        cout << "Jumlah Pintu: " << jumlahPintu << endl;
    }

    // Destructor
    ~Car()
    {


    }
};

File: Python/Program/Car.py
from Vehicle import Vehicle

class Car(Vehicle):
    def __init__(self, jumlah_kursi=0, jumlah_pintu=0, plat_nomor="", merk="", tahun_produksi="", warna=""):
        super().__init__(plat_nomor, merk, tahun_produksi, warna)  #panggil konstruktor kelas induk
        self.jumlah_kursi = jumlah_kursi  #inisialisasi atribut jumlah_kursi
        self.jumlah_pintu = jumlah_pintu  #inisialisasi atribut jumlah_pintu

    def set_jumlah_kursi(self, jumlah_kursi):  #metode untuk mengatur jumlah_kursi
        self.jumlah_kursi = jumlah_kursi

    def set_jumlah_pintu(self, jumlah_pintu):  #metode untuk mengatur jumlah_pintu
        self.jumlah_pintu = jumlah_pintu

    def get_jumlah_kursi(self):  #metode untuk mendapatkan jumlah_kursi
        return self.jumlah_kursi

    def get_jumlah_pintu(self):  #metode untuk mendapatkan jumlah_pintu
        return self.jumlah_pintu

    def display(self):  #metode untuk menampilkan informasi mobil
        super().display()  #panggil metode display dari kelas induk
        print("Jumlah Kursi:", self.jumlah_kursi)  #tampilkan jumlah_kursi
        print("Jumlah Pintu:", self.jumlah_pintu)  #tampilkan jumlah_pintu
